src/services/has-many-getter.js

```diff
@@ -22,10 +22,18 @@ function HasManyGetter(model, association, opts, params) {
 
   const fieldNamesRequested = getFieldNamesRequested();
   const searchBuilder = new SearchBuilder(association, opts, params, fieldNamesRequested);
-  const where = searchBuilder.perform(params.associationName);
   const include = queryBuilder.getIncludes(association, fieldNamesRequested);
 
-  function findQuery(queryOptions) {
+  let where;
+
+  async function getWhere() {
+    if (!where) {
+      where = await searchBuilder.perform(params.associationName);
+    }
+    return where;
+  }
```

Comment:
[IMPROVEMENT] I don't like that, the `where` value should initialized as the other, no need to create a function.

Reply:
It's async. So I cannot initialized it like the other 😢